src/unity.c

```diff
@@ -203,19 +188,21 @@ void UnityPrintNumberUnsigned(const UNITY_UINT number)
 /*-----------------------------------------------*/
 void UnityPrintNumberHex(const UNITY_UINT number, const char nibbles_to_print)
 {
-    UNITY_UINT nibble;
+    int nibble;
```

Comment:
Just doing the simplest thing... 
I've learned `UNITY_INT` is always 32-bits or 64-bits, and `int` matches the register size of the target platform. The value assigned is bounded in the range {0,15}, so any type could work. The sum below, `'0' + nibble`, becomes an int, too.